lib/shopping_cart.rb

```diff
@@ -0,0 +1,46 @@
+class ShoppingCart
+    attr_reader :name, :capacity, :products
+    def initialize(name, capacity)
+        @name = name
+        @capacity = capacity.to_i
+        @products = []
+        @full = false
+    end
+
+    def add_product(product)
+        @products << product
+    end
+
+    def details
+        cart_hash = {name: @name, capacity: @capacity}
+    end
+
+    def total_number_of_products
+        @products.inject(0) {|sum, product| sum += product.quantity}
```

Comment:
👍 Nice job trying out inject! The `sum` enumerable would also be a good one for this too.